IllegalStateException: Nesting problem

stacktrace:
```
Caused by: java.lang.IllegalStateException: Nesting problem.
        at com.google.gson.stream.JsonWriter.beforeName(JsonWriter.java:667) ~[gson-2.10.jar:?]
        at com.google.gson.stream.JsonWriter.writeDeferredName(JsonWriter.java:401) ~[gson-2.10.jar:?]
        at com.google.gson.stream.JsonWriter.value(JsonWriter.java:417) ~[gson-2.10.jar:?]
        at team.unnamed.creative.serialize.minecraft.SerializerBlockState.writeCondition(SerializerBlockState.java:198) ~[?:?]
        at team.unnamed.creative.serialize.minecraft.SerializerBlockState.writeCondition(SerializerBlockState.java:192) ~[?:?]
        at team.unnamed.creative.serialize.minecraft.SerializerBlockState.writeCondition(SerializerBlockState.java:227) ~[?:?]
        at team.unnamed.creative.serialize.minecraft.SerializerBlockState.writeCondition(SerializerBlockState.java:192) ~[?:?]
        at team.unnamed.creative.serialize.minecraft.SerializerBlockState.writeCondition(SerializerBlockState.java:210) ~[?:?]
        at team.unnamed.creative.serialize.minecraft.SerializerBlockState.writeSelector(SerializerBlockState.java:171) ~[?:?]
        at team.unnamed.creative.serialize.minecraft.SerializerBlockState.serialize(SerializerBlockState.java:68) ~[?:?]
        at team.unnamed.creative.serialize.minecraft.SerializerBlockState.serialize(SerializerBlockState.java:44) ~[?:?]
        at team.unnamed.creative.serialize.minecraft.MinecraftResourcePackWriterImpl.writeToJson(MinecraftResourcePackWriterImpl.java:121) ~[?:?]
        at team.unnamed.creative.serialize.minecraft.MinecraftResourcePackWriterImpl.write(MinecraftResourcePackWriterImpl.java:72) ~[?:?]
        at team.unnamed.creative.serialize.minecraft.MinecraftResourcePackWriter.build(MinecraftResourcePackWriter.java:79) ~[?:?]
        at ru.melonhell.resourcepackmerger.MergeService.build(MergeService.kt:46) ~[ResourcePackMerger-1.0.0-SNAPSHOT.jar:?]
```

siplified code:
```kotlin
val file = File("ia.zip")
val pack = MinecraftResourcePackReader.minecraft().readFromZipFile(file)
val builtPack = MinecraftResourcePackWriter.minecraft().build(pack) // exception on this line
```
ia.zip from example: [ia.zip](https://github.com/unnamed/creative/files/11545597/ia.zip)

--- a/serializer-minecraft/src/main/java/team/unnamed/creative/serialize/minecraft/SerializerBlockState.java
+++ b/serializer-minecraft/src/main/java/team/unnamed/creative/serialize/minecraft/SerializerBlockState.java
@@ -223,7 +223,7 @@ final class SerializerBlockState implements JsonFileStreamWriter<BlockState>, Js
 
             writer.name("AND").beginArray();
             for (Condition child : conditions) {
-                writer.beginArray();
+                writer.beginObject();
                 writeCondition(writer, child);
                 writer.endObject();
             }

